app/forms/bulk_update/bulk_add_trainees_upload_form.rb

```diff
@@ -22,14 +22,11 @@ def initialize(provider: nil, file: nil)
     def save
       return false unless valid?
 
-      BulkUpdate::TraineeUpload.create(
-        provider: provider,
-        file: file,
-        file_name: file.original_filename,
-        number_of_trainees: csv&.count,
-        status: valid? ? :pending : :failed,
-        error_messages: errors.messages.values.inject([], &:concat),
-      )
+      upload = create_upload
```

Comment:
I suggest to wrap this in a transaction so as to avoid impartial records to be created.